src/primitive/circle.rs

```diff
@@ -0,0 +1,201 @@
+//! Circle primitive
+
+use cgmath::{BaseFloat, Point2, Vector2};
+use cgmath::prelude::*;
+
+use {Aabb2, Ray2};
+use prelude::*;
+
+/// Circle primitive
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "eders", derive(Serialize, Deserialize))]
+pub struct Circle<S> {
+    /// Radius of the circle
+    pub radius: S,
+}
+
+impl<S> Circle<S> {
+    /// Create a new circle primitive
+    pub fn new(radius: S) -> Self {
+        Self { radius }
```

Comment:
this used to be unsupported syntax, since when is it stable?

Reply:
I have no idea :)